.github/workflows/rust.yml

```diff
@@ -109,14 +109,14 @@ jobs:
           export CARGO_HOME="/github/home/.cargo"
           export CARGO_TARGET_DIR="/github/home/target"
           cd rust
-          cargo test
+          # run tests on all workspace members
+          cargo test --all --features=prettyprint
           # test datafusion examples
           cd datafusion
           cargo run --example csv_sql
           cargo run --example parquet_sql
           cd ..
           cd arrow
-          cargo test
```

Comment:
this appears to run all the arrow tests twice -- once in the main workspace and once in the arrow crate again. I am not sure that is adding any value

Reply:
This was introduced when I was trying to test the SIMD and non-SIMD versions of the code.  Feature options were not being respected within the sub-crates when supplied at the workspace level, see [here](https://github.com/rust-lang/cargo/issues/5015#issuecomment-532483775).  So the solution was to move into the sub-crate to enable/disable the feature.

Quickly reading through the cargo issues, it's not clear that this has been resolved yet.

Reply:
AH, that makes sense @paddyhoran  -- I think the feature flag test for simd got moved to an entire different job, namely the   `linux-test-simd` job below. I think I need to re-instante the second CI test but run the tests with the pretty print option

Reply:
In fact the CI test failed, exactly as you predicted https://github.com/apache/arrow/runs/1727917047 -- namely that you can't specify `--features` in the root of the workspace:

```
Run export CARGO_HOME="/github/home/.cargo"
error: --features is not allowed in the root of a virtual workspace
note: while this was previously accepted, it didn't actually do anything
help: change the current directory to the package directory, or use the --manifest-path flag to the path of the package
Error: Process completed with exit code 101.
```

Reply:
A solution could be to exclude arrow from the initial `cargo test`, then test it after `cd arrow`, with the prettyprint (and other future) feature.

Or has this been resolved?

Reply:
It has been resolved (though the resolution was to keep running the tests for arrow twice)

Reply:
To be fair, running the arrow tests is a second or so, so the harm is pretty small. Re-compiling DataFusion's dependencies would have been a different beast ^_^

Reply:
Is everyone ok if I merge this in I think it will have minimal effect on overall test times (will require arrow to get recompiled one additional time I think)